forc-plugins/forc-debug/src/server/handlers/handle_launch.rs

```diff
@@ -0,0 +1,202 @@
+use crate::server::{AdapterError, DapServer};
+use crate::types::Instruction;
+use forc_pkg::{self, BuildProfile, Built, BuiltPackage, PackageManifestFile};
+use forc_test::execute::TestExecutor;
+use forc_test::setup::TestSetup;
+use forc_test::BuiltTests;
+use std::{collections::HashMap, fs, sync::Arc};
+use sway_types::span::Position;
+
+impl DapServer {
+    /// Handles a `launch` request. Returns true if the server should continue running.
+    pub fn handle_launch(&mut self) -> Result<bool, AdapterError> {
+        // Build tests for the given path.
+        let (pkg_to_debug, test_setup) = self.build_tests()?;
+        let entries = pkg_to_debug.bytecode.entries.iter().filter_map(|entry| {
+            if let Some(test_entry) = entry.kind.test() {
+                return Some((entry, test_entry));
+            }
+            None
+        });
+
+        // Construct a TestExecutor for each test and store it
+        let executors: Vec<TestExecutor> = entries
+            .filter_map(|(entry, test_entry)| {
+                let offset = u32::try_from(entry.finalized.imm)
+                    .expect("test instruction offset out of range");
+                let name = entry.finalized.fn_name.clone();
+                if test_entry.file_path.as_path() != self.state.program_path.as_path() {
+                    return None;
+                }
+
+                Some(TestExecutor::new(
+                    &pkg_to_debug.bytecode.bytes,
+                    offset,
+                    test_setup.clone(),
+                    test_entry,
+                    name.clone(),
+                ))
+            })
+            .collect();
+        self.state.init_executors(executors);
+
+        // Start debugging
+        self.start_debugging_tests(false)
+    }
+
+    /// Builds the tests at the given [PathBuf] and stores the source maps.
+    pub(crate) fn build_tests(&mut self) -> Result<(BuiltPackage, TestSetup), AdapterError> {
```

Comment:
It is interesting to see that we need this much code to build something with tests enabled 🤔 

Maybe we need to refactor forc-pkg a little sometime to provide easier to use APIs for such cases.

Reply:
Also we should visit this function some time later because cloning `BuiltPackage` is a costly event that might lead to huge memory usage stuff that we encountered before. I think we should be fine for now but maybe in a separate PR that refactors `forc-pkg` as mentioned above, we can try to remove all/some clone's on built package's across the repo.

Reply:
Great call out. I think the interface of forc-pkg can definitely be improved, but I didn't want to bloat this PR too much. Opened https://github.com/FuelLabs/sway/issues/5620